vms/platformvm/blocks/builder/builder.go

```diff
@@ -392,6 +402,25 @@ func buildBlock(
 		)
 	}
 
+	// TODO@ array of deposits with equal timestamp
+	depositTxID, shouldUnlock, err := builder.getNextDepositToUnlock(timestamp, parentState)
```

Comment:
I think we have to compare staker time and deposit time, and use the smaller one.
This impl. priorizes always staker

Reply:
I intended to do as u said, probably just didn't implement yet